runtime/io/io.cpp

```diff
@@ -0,0 +1,389 @@
+#include <gmp.h>
+#include <cstring>
+#include <stdexcept>
+
+#include "fcntl.h"
+#include "unistd.h"
+#include "runtime/alloc.h"
+#include "runtime/header.h"
+
+extern "C" {
+
+#define KCHAR char
+#define ERRTAG(err) "Lbl'hash'" #err
+
+  mpz_ptr move_int(mpz_t);
+
+  static block dotK = {getBlockHeaderForSymbol((((uint64_t)getTagForSymbolName("dotk")) << 32) | 1), {}};
+  static blockheader kseqHeader = {getBlockHeaderForSymbol((uint64_t)getTagForSymbolName("kseq{}"))};
+
+  block * block_errno() {
+    block * retBlock = static_cast<block *>(koreAlloc(sizeof(block)));
+    switch (errno) {
+    case EOF:
```

Comment:
create a local char * and assign ERRTAG(EOF) to it, then do the rest after the switch statement. Also, don't forget to break. Finally, put the case and the statement and the break on one line to make it easier to read.